### PR TITLE
Replace juju_config.GA_KEY with juju_config.GTM_enabled

### PR DESCRIPTION
**Pull request by anthonydillon**

### CODE REVIEW

File: jujugui/options.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ def update(settings):
     _update(settings, 'jujugui.base_url', default=None)
     _update(settings, 'jujugui.charmstore_url', default=DEFAULT_CHARMSTORE_URL)
     _update(settings, 'jujugui.api_path', default=DEFAULT_API_PATH)
-    _update(settings, 'jujugui.ga_key', default='')
+    _update(settings, 'jujugui.GTM_enabled', default='')
```
Comment (jaycee) [inline jujugui/options.py]: Default here should be 'False', as we want this to be a boolean setting.

Comment (hatched) [inline jujugui/options.py]: Should this not have a boolean default value? `False`

File: jujugui/static/gui/src/app/config-debug.js [hunk 2]
```diff
@@ -53,10 +53,10 @@ var juju_config = { // eslint-disable-line no-unused-vars
...
   simulateEvents: false,
   readOnly: false,
-  // Set the GA_key to enable Google Analytics usage and calls. Also implies
-  // using cookies. For the debug configuration, the GA_key should be blank to
-  // prevent muddying the gathered statistics.
-  GA_key: '',
+  // Set the GTM_enabled to enable Google Tag Manager usage and calls. Also
+  // implies using cookies. For the debug configuration, the GTM_enabled should
+  // be false to prevent muddying the gathered statistics.
+  GTM_enabled: false,
```
Comment (jaycee) [inline jujugui/static/gui/src/app/config-debug.js]: I think we can actually remove config-debug.js and config-prod.js, can't we? @hatched?

File: jujugui/tests/test_options.py [hunk 3]
```diff
@@ -15,7 +15,7 @@ class TestUpdate(unittest.TestCase):
         'jujugui.base_url': None,
         'jujugui.charmstore_url': options.DEFAULT_CHARMSTORE_URL,
         'jujugui.combine': True,
-        'jujugui.ga_key': '',
+        'jujugui.GTM_enabled': '',
```
Comment (jaycee) [inline jujugui/tests/test_options.py]: Again, as this is defaults we should do False, not ''.

File: jujugui/tests/test_views.py [hunk 4]
```diff
@@ -98,7 +98,7 @@ def test_default_options(self):
         self.assertEqual('wss', config['socket_protocol'])
         self.assertEqual(
             options.DEFAULT_CHARMSTORE_URL, config['charmstoreURL'])
-        self.assertEqual('', config['GA_key'])
+        self.assertEqual('', config['GTM_enabled'])
```
Comment (jaycee) [inline jujugui/tests/test_views.py]: And then self.assertFalse(config['GTM_enabled'])

File: jujugui/static/gui/src/app/config-debug.js (hunk 2 above)
Comment (hatched) [inline jujugui/static/gui/src/app/config-debug.js]: That's correct - although out of scope for this branch.

**Comment by jaycee** [comment]
@anthonydillon thanks, just a few fixes so we're using boolean values instead of empty strings.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4272/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4273/
Test PASSed.

File: jujugui/static/gui/src/app/app.js [hunk 5]
```diff
@@ -2067,8 +2067,8 @@ YUI.add('juju-gui', function(Y) {
      *
      */
     authorizeCookieUse: function(req, res, next) {
-      var ga_key = this.get('GA_key');
-      if (ga_key) {
+      var GTM_enabled = this.get('GTM_enabled');
```
Comment (jaycee) [inline jujugui/static/gui/src/app/app.js]: I'm not sure this is going to work--GTM_enabled isn't an attr on app, so this.get shouldn't work.

That said GA_key wasn't an attr either--I'm not sure how this worked previously. @hatched, you have any ideas?

Comment (hatched) [inline jujugui/static/gui/src/app/app.js]: Anything that's set in `juju_config` will get applied to app.js so this should work.
https://github.com/juju/juju-gui/blob/develop/jujugui/templates/index.html.mako#L398

**Comment by hatched** [comment]
:+1: :ok:

When setting `jujugui.GTM_enabled` in `development.ini` the cookie notification was shown the first page load and the GTM script was executed and additional scripts were downloaded.

**Comment by jaycee** [comment]
:+1: thanks @anthonydillon

**Comment by anthonydillon** [comment]
Thank you :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**